docs/DeveloperGuide.md

```diff
@@ -1,11 +1,308 @@
+the `ParserManager` passes the commands on to the respective exceptions parsers which inherit from the `ExceptionsParser` interface. 
+The exceptions parsers consist of:  
+- `ProjectParser`
+- `MemberParser`
+- `TaskParser`
+- `SprintParser`
+- `HelpParser`
+
+The `ProjectParser` validates the parameters of the command. If the command is valid, it returns the respective `ProjectCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `ProjectParser` returns an appropriate warning message to the user.  
+
+The `MemberParser` validates the parameters of the command. If the command is valid, it returns the respective `MemberCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `TaskParser` validates the parameters of the command. If the command is valid, it returns the respective `TaskCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `TaskParser` returns an appropriate warning message to the user.  
+
+The `SprintParser` validates the parameters of the command. If the command is valid, it returns the respective `SprintCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `SprintParser` returns an appropriate warning message to the user.  
+
+The `HelpParser` validates the parameters of the command. If the command is valid, it returns the respective `HelpParserCommandXYZ` to the `ParserManager`.
+If the command is invalid, the `HelpParser` returns an appropriate warning message to the user.
+
+
+The subcommand classes `HelpParserCommandXYZ`, `ProjectCommandXYZ`, `MemberCommandXYZ`, `TaskCommandXYZ`, `SprintCommandXYZ` all inherit from an abstract `Command` class, 
+which has an execute function.
+
+The `ParserManager` then returns the command back to `SCRUMptious`, which then executes the command.
+
+A detailed list of the subcommand classes is as follows:
+* Project Commands:  
+    * `CreateProjectCommand`
+    * `ListProjectCommand`
+    * `SelectProjectCommand`
+    * `ViewProjectCommand`
+* Member Commands:
+    * `AddMemberCommand`
+    * `DeleteMemberCommand`
+* Task Commands:
+    * `AddTaskCommand`
+    * `ChangeTaskPriorityCommand`
+    * `DeleteTaskCommand`
+    * `DoneTaskCommand`
+    * `ViewTaskCommand`
+* Sprint Commands:
+    * `AddSprintTaskCommand`
+    * `AllocateSprintTaskCommand`
+    * `CreateSprintCommand`
+    * `DeallocateSprintTaskCommand`
+    * `EditSprintCommand`
+    * `RemoveSprintTaskCommand`
+    * `ViewSprintCommand`
+* Help Commands:
+    * `HelpCommand`
+    * `MemberHelpCommand`
+    * `ProjectHelpCommand`
+    * `SprintHelpCommand`
+    * `StorageHelpCommand`
+    * `TaskHelpCommand`
+### Model Component
+{UML}
+### Storage Component
+![Figure X: Simplified class diagram for Storage Component, Model and json.simple](./image/developerguide/storagecomponent.png "Storage Component UML")  
+  
```

Comment:
![image](https://user-images.githubusercontent.com/52351742/97386218-54ed3380-190e-11eb-84e8-1ed024ea511d.png)

Perhaps you could add association labels to make the diagram more readable for the user.